src/vs/workbench/api/common/extHostLanguageFeatures.ts

```diff
@@ -423,6 +424,13 @@ class CodeActionAdapter {
 			if (!candidate) {
 				continue;
 			}
+
+			const currentKind = (candidate as vscode.CodeAction).kind;
```

Comment:
While this cast will likely work, it would be better to avoid it. Can you only run this code after checking that the `candidate` is not a command?